### PR TITLE
142 linking to contracts

### LINKED ISSUE #142: Linking to contracts
## Description
As a customer I want to be able to share links to specific contracts so that I can show them to my colleagues, and so that I can save them as a bookmark.

## Acceptance criteria
- [x] The URL in the browser reflects the currently open contract
- [x] The contract detail view URL is `/contracts/{contractId}`
- [x] When the page is refreshed, the details view remains open
- [x] When clicking a link to a specific contract, the detail view of that contract is opened

### PR DESCRIPTION
**Pull request by martinjonsson01**
## Summary
Makes contracts linkable by updating the URL dynamically when navigating to contracts.

chrome_KWBEAyiqhO

## Definition of done
- [x] All acceptance criteria are completed.
- [x] The code has (at least) 80% branch test coverage.
- [x] All tests pass.
- [x] The code is reviewed and accepted by (at least) two other team members.
- [x] Views adhere to design style.

Closes #142

### CODE REVIEW

File: src/Client/Pages/Contracts/ContractDetails.razor [hunk 1]
```diff
@@ -1,9 +1,13 @@
...
-@inject ISessionService _session
+@implements IAsyncDisposable
+@inject ISessionService Session
+@inject IJSRuntime Js
+@inject HttpClient Http
+@inject NavigationManager NavigationManager
 
 <!-- modal-dialog-scrollable -->
-<div class="modal w-100" id="@("id_" + Contract.Id)">
+<div class="modal w-100" id="@ModalId">
```
Comment (JacobBredin) [inline src/Client/Pages/Contracts/ContractDetails.razor:10]: Nice

File: src/Client/Pages/Contracts/FavoriteButton.razor [hunk 2]
```diff
@@ -3,11 +3,15 @@
...
+@inject HttpClient Http
+@inject ISessionService Session
 
-<button id="favorite-button" class="btn p-0 white-text" @onclick="ChangeFavorite">
-    @if (isFavorite)
+<button id="favorite-button"
+        class="btn p-0 white-text"
+        @onclick="ChangeFavorite"
+        @onclick:stopPropagation="true"
+        @onclick:preventDefault="true">
```
Comment (JacobBredin) [inline src/Client/Pages/Contracts/FavoriteButton.razor:13]: nice 🤩

**Review by JacobBredin — APPROVED** [review]
Nice 🥳

**Review by Matte02 — APPROVED** [review]
Lgtm 👍

**Pull request merged**